### PR TITLE
PHOENIX-5117 : Return the count of rows scanned in HBase

### PR DESCRIPTION
**Pull request by palashc**
Trying to apply patch attached to the jira - https://issues.apache.org/jira/browse/PHOENIX-5117

### CODE REVIEW

**Comment by ranganathg** [comment]
This earlier PR #1584 from @gtaparia can be closed?

File: phoenix-core-client/src/main/java/org/apache/phoenix/iterate/OrderedResultIterator.java [hunk 1]
```diff
@@ -482,6 +486,13 @@ private void getDummyResult() {
         }
     }
 
+    public ScannerContext getRegionScannerContext() {
+        if (hasRegionScannerContext) {
+            return ((RegionScannerResultIterator)delegate).getRegionScannerContext();
```
Comment (ranganathg) [inline phoenix-core-client/src/main/java/org/apache/phoenix/iterate/OrderedResultIterator.java:491]: `((RegionScannerResultIterator)getDelegate()).getRegionScannerContext();` ? to be consistent with the code change you did in [OffsetResultIterator.java](https://github.com/apache/phoenix/pull/1918/files#diff-bc6d903dfea2e0dd398c19319778e10a4cca04ab7006f0a91a2d1b06da5aa4e0)

**Comment by gourabtaparia** [comment]
> This earlier PR #1584 from @gtaparia can be closed?

I have closed the PR @ranganathg

File: phoenix-core-client/src/main/java/org/apache/phoenix/iterate/OrderedResultIterator.java (hunk 1 above)
Comment (palashc) [reply phoenix-core-client/src/main/java/org/apache/phoenix/iterate/OrderedResultIterator.java:491]: @ranganathg Sure that makes sense. Note that this change is still a WIP, I have not been able to apply the patch properly just yet. For eg, the changes in `OrderedResultIterator` and `OffsetResultIterator.java` introduce a dependency in phoenix-core-client on phoenix-core-server which leads to a cyclic dependency.

**Pull request closed without merging**